tests/cases/compiler/assignmentToAnyArrayRestParameters.ts

```diff
@@ -12,3 +12,12 @@ function foo<T extends string[]>(
     const f3: (...args: any) => string = fb;
     const f4: (...args: any[]) => string = fb;
 }
+
+function bar<T extends string[], K extends number>() {
+    type T00 = string[]["0"];
+    type T01 = string[]["0.0"];  // Error
+    type T02 = string[][K | "0"];
+    type T10 = T["0"];
+    type T11 = T["0.0"];  // Error
+    type T12 = T[K | "0"];
+}

```

Comment:
nit: those should likely be in a different test file since they don't have anything in common with this "test title" (`assignmentToAnyArrayRestParameters`)